cilium/cmd/bpf_lb_maglev_list.go

```diff
@@ -21,52 +22,59 @@ var bpfMaglevListCmd = &cobra.Command{
 	Run: func(cmd *cobra.Command, args []string) {
 		common.RequireRootPrivilege("cilium bpf lb maglev list")
 
-		lookupTables := map[string][]string{}
-		dumpMaglevTables(lookupTables)
+		backends, err := dumpMaglevTables()
+		if err != nil {
+			Fatalf("Unable to dump Maglev lookup tables: %s", err)
+		}
 
 		if command.OutputJSON() {
-			if err := command.PrintOutput(lookupTables); err != nil {
+			if err := command.PrintOutput(backends); err != nil {
 				Fatalf("Unable to generate JSON output: %s", err)
 			}
 			return
 		}
 
-		TablePrinter("SVC ID", "LOOKUP TABLE", lookupTables)
+		TablePrinter("SVC ID", "LOOKUP TABLE", backends)
 	},
 }
 
-func parseMaglevEntry(key *lbmap.MaglevOuterKey, value *lbmap.MaglevOuterVal, tableSize uint32, tables map[string][]string) {
-	innerMap, err := lbmap.MaglevInnerMapFromID(int(value.FD), tableSize)
+// dumpMaglevTables returns the contents of the Maglev v4 and v6 maps
+// in a format the table printer expects.
+func dumpMaglevTables() (map[string][]string, error) {
+	out, err := dumpMaglevTable(lbmap.MaglevOuter4MapName)
 	if err != nil {
-		Fatalf("Unable to get map fd by id %d: %s", value.FD, err)
+		return nil, err
 	}
 
-	innerKey := lbmap.MaglevInnerKey{
-		Zero: 0,
-	}
-	innerValue, err := innerMap.Lookup(&innerKey)
+	v6, err := dumpMaglevTable(lbmap.MaglevOuter6MapName)
 	if err != nil {
-		Fatalf("Unable to lookup element in map by fd %d: %s", value.FD, err)
+		return nil, err
 	}
 
-	tables[fmt.Sprintf("%d", key.ToNetwork().RevNatID)] = []string{fmt.Sprintf("%v", innerValue.BackendIDs)}
+	// Merge v6 lookup tables into v4 result.
+	// Overlapping entries will clobber each other.
```

Comment:
They should not overlap, as it's not possible that two services will end up with the same ID (even if they belong to different IP families).